10.0 add module

sebastienbeau opened the pull request.

sebastienbeau committed:
[REF] rename module

sebastienbeau committed:
[REF] start to refactor the module. Simplify the call to odoo by using simple json-rpc api like. Add test with a mock based on file. Start to reimplement message support

sebastienbeau committed:
[REF] start to make working the message

sebastienbeau committed:
[IMP] finish post message, manage author correctly with an automatic synchronisation between the ERP. Hide useless stuff of the client chatter. Do not return internal note (keep it private)

sebastienbeau committed:
[IMP] add author id on task, set the right partner on tracking notification, synchronize email and mobile, phone on partner, fix test and update test data

sebastienbeau committed:
[REF] refactor code for mapping the partner author and assign on the task

sebastienbeau committed:
[IMP] implmeent correctly the support of url for the task. We need to build a full url as the server do not have access from client data

sebastienbeau committed:
[IMP] add project multi support, now the auth key is on the customer, and we can show all customer project

sebastienbeau committed:
[IMP] add filter for project in view

sebastienbeau committed:
[REF] refactor and remove ueless code

sebastienbeau committed:
[IMP] add the posibility to select the project on the task

sebastienbeau committed:
[IMP] add code to send an email when the task is assigned

sebastienbeau committed:
[IMP] set nobody for assign by default and set a default author_id. Also add correct domain for author_id

sebastienbeau committed:
[IMP] automatically create partner when receive an email

sebastienbeau committed:
[IMP] update customer contact if the email match, and solve issue if a partner have been delete and the message still exist

sebastienbeau committed:
[REF] move code to assign mail in a separated module, fix autosubcribe, add the body and the right author when processing an incomming email

sebastienbeau committed:
[IMP] fix install/update and test

sebastienbeau committed:
[IMP] improve kanban view

sebastienbeau committed:
[PEP] clean pep8

sebastienbeau committed:
[FIX] fix travis

sebastienbeau committed:
[IMP] add the support of partner created from email

sebastienbeau committed:
[IMP] add error management, solve view issue for removing note button, add customer_link

sebastienbeau committed:
[IMP] add translation

sebastienbeau committed:
[FIX] fix test

sebastienbeau committed:
[REF] refactor the read group method and always show all the stage

sebastienbeau committed:
[IMP] add default filter for project, fix order of project, add uniq constraint to avoid to have duplicated partner (multi-read in parallel)

sebastienbeau committed:
[IMP] hide all useless field at the creation

sebastienbeau committed:
[FIX] fix wrong xmlid

sebastienbeau committed:
[FIX] allow partner creation with missing image

PierrickBrun committed:
fix demo data

bealdav committed:
FIX proj_api_cli: exclude menu support in dev

IMP proj_api_cli: escape with keychain api

[FIX] do nothing if no keychain account for this environment

sebastienbeau committed:
[IMP] do not raise an error when loading the menu if odoo server is down

sebastienbeau committed:
[IMP] Add two group support and basic user for support, adapt the view

sebastienbeau committed:
[IMP] add the possibility to change the assignee

sebastienbeau committed:
[IMP] fix filter on assignee, add filter on my task

sebastienbeau committed:
[IMP] add the possility to assign a ticket to the customer

sebastienbeau committed:
[IMP] track visibility of assignee to customer and add him in the follower

sebastienbeau committed:
[IMP] update data

bguillot committed:
[IMP] project_api module with attachment and tags synchronization

bguillot committed:
[IMP] sync attachmentand tags in project_api_client

bguillot committed:
[FIX] tag and color

bguillot committed:
[FIX] fix project_api_client

sebastienbeau committed:
[FIX] fix issue with menu, we should return browse records

bguillot committed:
[FIX] add functional_area field in project api

bguillot committed:
[FIX] clean with black

bguillot committed:
[IMP] add option for automaticaly unsubscribe other users when chosse an user on task

bguillot committed:
[IMP] add pre-commit

sebastienbeau committed:
[IMP] add customer report

sebastienbeau committed:
[IMP] show customer report

mourad-ehm committed:
[FIX] black formating

mourad-ehm committed:
[FIX] stock_reserve_location from test & fix project_api_client view bug

mourad-ehm committed:
[FIX] fix pylint on label_wizard pricelist_tax product_supplier_intercompany production_release project_api

mourad-ehm committed:
[FIX] split attachment_service and task into 2 file & fix Unreachable code

mourad-ehm committed:
[FIX] attachment_service call super read method

mourad-ehm committed:
[FIX] pricelist_tax travis test due to dependency collusion

bealdav committed:
IMP project_api: partner view

bguillot committed:
[FIX] kanban view and stop use of keychain

sebastienbeau committed:
[IMP] fix black

sebastienbeau committed:
[FIX] add oca dependency

sebastienbeau committed:
fixup! [INIT] init project

sebastienbeau committed:
Black me with OCA config

sebastienbeau committed:
fixup! [INIT] init project

sebastienbeau committed:
[REF] refactor code, backport change done by David, fix code and do some extra refactor

sebastienbeau committed:
[IMP] improve UI and add cache for reading support key

sebastienbeau committed:
[IMP] give the possibility to restrict users for assigning internal ticket

sebastienbeau committed:
[IMP] improve UI and add rename field

sebastienbeau committed:
[IMP] add a process for testing the connection and initial sync of support partner

sebastienbeau committed:
[IMP] improve management of project list and type to store them locally and avoid useless call

bealdav commented on project_api/__manifest__.py:
```diff
@@ -0,0 +1,19 @@
+
+{
+    "name": "Project API",
+    "summary": "Expose a json-rpx like API on top of base_rest",
```
```suggestion
    "summary": "Expose a json-rpc like API on top of base_rest",
```

bealdav commented on project_api_client/models/action.py:
```diff
@@ -0,0 +1,46 @@
+        return {
+            "id": action.id,
+            "name": _("Create a Ticket"),
+            "res_model": u"external.task",
```
```suggestion
            "res_model": "external.task",
```

bealdav commented on project_api_client/models/action.py:
```diff
@@ -0,0 +1,46 @@
+            "name": _("Create a Ticket"),
+            "res_model": u"external.task",
+            "src_model": model,
+            "type": u"ir.actions.act_window",
```
```suggestion
            "type": "ir.actions.act_window",
```

bealdav commented on project_api_client/models/support_account.py:
```diff
@@ -0,0 +1,112 @@
+        return self.browse(self._get_id_for_company(self.env.user.company_id.id))
+
+    @tools.ormcache("company_id")
+    def _get_id_for_company(self, company_id):
```
Peux-tu mettre une docstring pour expliquer l'intéret de ce décorateur en général et/ou dans ce cas. Peu de personnes le maitrise je pense. Merci

bealdav commented on .travis.yml:
```diff
@@ -25,7 +25,7 @@ jobs:
     - stage: linting
       name: "pre-commit"
-      python: "2.7"
+      python: "3.6"
```
Aie, v10 ne tourne pas avec python 3 !!!

bealdav reviewed (changes requested):
partial review

sebastienbeau committed:
[IMP] Improve usability

- split assignee in two field assignee_supplier_id and
assignee_customer_id
- fix missing kanban report
- give the posibility to set the project, assignee_customer_id at creation

sebastienbeau committed:
[FIX] re-record test and fix them

sebastienbeau committed:
Apply suggestions from code david review

Co-Authored-By: David Beal <david.beal@akretion.com>

sebastienbeau committed:
[FIX] fix travis config

rvalyi commented on project_api/README.rst:
```diff
@@ -0,0 +1,16 @@
+Project API
+==============
+
+This module add a restricted json-rpc like API for exposing your project customer
```
s/add/adds/

rvalyi commented on project_api_client/README.rst:
```diff
@@ -0,0 +1,37 @@
+Project API client
+=========================
+
+This module add a new entry "support" to allow to interact with your integrator
```
s/add/adds/
s/new entry/new menu entry/
s/to allow to/to allow you to/

rvalyi reviewed (changes requested):
the global README is also very bad English, I propose instead:
```

These modules are used and developped by Akretion to support our customers.

Basically we have 2 modules:

* project_api that exposes an API in our ERP and adds some extra features
* project_api_client that is installed in our customer ERP's and interact with our ERP through the API
```

sebastienbeau committed:
[FIX] fix comment and travis

sebastienbeau committed:
[FIX] fix summary
